src/main/java/ProjectOcean/Controller/StudyPlansController.java

```diff
@@ -0,0 +1,116 @@
+        this.showStudyPlan = showStudyPlan;
+
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(
+                "/StudyPlansWindow.fxml"));
+        fxmlLoader.setRoot(this);
+        fxmlLoader.setController(this);
+
+        try {
+            fxmlLoader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+
+        displayAllStudyPlans();
+
+    }
+
+    @FXML
+    private void onAddStudyPlanClicked(MouseEvent event) {
+
+        deactivateCurrStudyPlanButton();
+        model.addStudyPlan(); // Method also setting new studyPlan to current.
+
+        displayAllStudyPlans();
+        showStudyPlan.showAStudyPlan();
+
+        event.consume();
+
+    }
+
+    private void displayAllStudyPlans() {
+        studyPlanContainer.getChildren().clear();
+        studyPlanContainer.getChildren().add(addButton);
+
+        for (StudyPlan studyPlan : model.getStudent().getAllStudyPlans()) {
+            // Set new buttons properties, values taken from button created in ApplicationWindow.fxml
+            int nOfStudyPlans = studyPlanContainer.getChildren().size();
+            ButtonController newButton = new ButtonController(model, showStudyPlan, this::deactivateCurrStudyPlanButton, nOfStudyPlans, studyPlan);
+
+            studyPlanContainer.getChildren().add(nOfStudyPlans - 1, newButton);
+
+            if (newButton.getStudyPlan() == model.getStudent().getCurrentStudyPlan()) {
+                newButton.activateDefaultButton();
+            }
+        }
+    }
+
+    private void deactivateCurrStudyPlanButton() {
+        ButtonController buttonController = getCurrStudyPlansButtonController();
+        if (buttonController != null) {
+            buttonController.deActivateDefaultButton();
+        }
+    }
+
+    // Premise: there is an active study plan button.
+    private ButtonController getCurrStudyPlansButtonController() {
+        ButtonController bc = null;
+        for (int i = 0; i < model.getStudent().getAllStudyPlans().size(); i++) {
+            ButtonController buttonController = (ButtonController) studyPlanContainer.getChildren().get(i);
+            if (buttonController.getStudyPlan() == model.getStudent().getCurrentStudyPlan()) {
+                bc = buttonController;
+            }
+        }
+        return bc;
+    }
+
+    /**
+     * Removing current study plan and show another one afterwards
+     */
+    public void deleteCurrentStudyPlan() {
+        if (model.getStudent().getAllStudyPlans().size() == 0) {
+            new Alert(Alert.AlertType.ERROR, "Warning: you have no study plan to delete!").showAndWait();
+        } else {
+            ButtonController buttonController = getCurrStudyPlansButtonController();
+            model.getStudent().removeStudyPlan(buttonController.getStudyPlan());
+            setCurrentStudyPlan();
+            displayAllStudyPlans();
+        }
+    }
+
+    private void setCurrentStudyPlan() {
```

Comment:
Says setCurrent, but all you do is set the first one to current, should the name not reflect this?
Also what happens if if-statement fails?